robonomics_liability/src/robonomics_liability/executor.py

```diff
@@ -46,8 +46,9 @@ def incoming_liability(msg):
 
         def finish_liability(msg):
             self.liability_finish = True
```

Comment:
Possible race condition because lock flag is set before data that used after lock.